### PR TITLE
[CI] Revise R tests.

### PR DESCRIPTION
**Pull request by trivialfis**
- Use the standard package check (check on the tarball instead of the source tree).
- Run commands in parallel.
- Cleanup dependencies installation.
- Replace makefile.
- Documentation.
- Test using the image from rhub.

Close https://github.com/dmlc/xgboost/issues/8429 .

### CODE REVIEW

File: .github/workflows/r_rhub.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    - name: Run R tests
+      shell: bash
+      run: |
+        cd R-package && \
+        /tmp/R-devel/bin/R CMD INSTALL . && \
+        /tmp/R-devel/bin/R -q -e "library(testthat); setwd('tests'); source('testthat.R')"
+
+  test-Rhub-debian:
+    # We use rhub to launch container here
```
Comment (trivialfis) [inline .github/workflows/r_rhub.yml:42]: remove

File: R-package/tests/helper_scripts/install_deps.R [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  "methods",
+  "data.table",
+  "jsonlite"
+)
+
+ncpus <- parallel::detectCores()
+print(paste0("Using ", ncpus, " cores to install dependencies."))
+
+if (.Platform$OS.type == "unix") {
+  print("Installing source package on unix.")
```
Comment (trivialfis) [inline R-package/tests/helper_scripts/install_deps.R:32]: typo.

File: R-package/tests/helper_scripts/install_deps.R [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  install.packages(
+    pkgs,
+    repo = "http://cloud.r-project.org",
+    dependencies = c("Depends", "Imports", "LinkingTo"),
+    Ncpus = parallel::detectCores()
+  )
+} else {
+  print("Installing binary package on Windows.")
```
Comment (trivialfis) [inline R-package/tests/helper_scripts/install_deps.R:40]: typo.

File: .github/workflows/r_rhub.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
+# Run expensive R tests with the help of rhub. Only triggered by a pull request review
+# See discussion at https://github.com/dmlc/xgboost/pull/6378
+
+name: XGBoost-R-Rhub
+
+on:
+  pull_request_review_comment:
```
Comment (trivialfis) [inline .github/workflows/r_rhub.yml:7]: /gha run r-hub-test

Comment (trivialfis) [reply .github/workflows/r_rhub.yml:7]: /gha run r-hub-test

Comment (trivialfis) [reply .github/workflows/r_rhub.yml:7]: /gha run rhub-test

Comment (trivialfis) [reply .github/workflows/r_rhub.yml:7]: /gha run rhub-test

Comment (trivialfis) [reply .github/workflows/r_rhub.yml:7]: /gha run rhub-test

Comment (trivialfis) [reply .github/workflows/r_rhub.yml]: /gha run r-nold-test

Comment (trivialfis) [reply .github/workflows/r_rhub.yml]: /gha run r-nold-test

Comment (trivialfis) [reply .github/workflows/r_rhub.yml]: /gha run r-nold-test

**Review by hcho3 — CHANGES_REQUESTED** [review]

**Review by hcho3 — APPROVED** [review]

**Pull request merged**